src/platform/src/main/java/org/geoserver/platform/GeoServerExtensions.java

```diff
@@ -161,12 +159,9 @@ public void setApplicationContext(ApplicationContext context)
         
         // load from factory spi
         List<Object> spiExtensions = spiCache.get(extensionPoint);
-        if(spiExtensions == null) {
+        if (spiExtensions == null) {
             spiExtensions = new ArrayList<Object>();
-            Iterator i = FactoryRegistry.lookupProviders(extensionPoint);
-            while( i.hasNext() ) {
-                spiExtensions.add( i.next() );
-            }
+            ServiceLoader.load(extensionPoint).iterator().forEachRemaining(spiExtensions::add);
```

Comment:
Okay this method was part of the ServiceRegistry (former superclass). ServiceLoader is the new correct way to use SPI discouvery.